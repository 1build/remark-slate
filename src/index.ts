import plugin, { transform, defaultNodeTypes } from './serializer';

export { transform, defaultNodeTypes };

export default plugin;
