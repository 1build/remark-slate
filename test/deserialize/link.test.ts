import { deserialize } from '../../src';

it('deserialize link', () => {
  expect(
    deserialize({
      type: 'link',
      url: 'https://jackhanford.com',
      children: [{ value: 'Personal website' }],
    })
  ).toMatchSnapshot();
});

it('deserialize link with linkDestinationKey', () => {
  expect(
    deserialize(
      {
        type: 'link',
        url: 'https://eaze.com',
        children: [{ value: 'Shopping' }],
      },
      { linkDestinationKey: 'custom-value' }
    )
  ).toMatchSnapshot();
});
