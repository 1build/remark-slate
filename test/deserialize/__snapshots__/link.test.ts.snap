// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deserialize link 1`] = `
Object {
  "children": Array [
    Object {
      "text": "Personal website",
    },
  ],
  "link": "https://jackhanford.com",
  "type": "link",
}
`;

exports[`deserialize link with linkDestinationKey 1`] = `
Object {
  "children": Array [
    Object {
      "text": "Shopping",
    },
  ],
  "custom-value": "https://eaze.com",
  "type": "link",
}
`;
